BridgeSDK/Networking/SBBNetworkManager.m

```diff
@@ -723,6 +723,26 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
 
 - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error
 {
+    if (error) {
+        NSData *resumeData = error.userInfo[NSURLSessionDownloadTaskResumeData];
+        if (resumeData) {
+            // there's resume data from a download task, so we'll use it to retry
+            NSURLSessionDownloadTask *resumeTask = [session downloadTaskWithResumeData:resumeData];
+            NSURLSessionDownloadTask *dlTask = (NSURLSessionDownloadTask *)task;
+
+            // transfer the information associated with the old task to the new one
+            resumeTask.taskDescription = task.taskDescription;
+            SBBNetworkManagerTaskCompletionBlock taskCompletion = [self completionBlockForTask:dlTask];
+            SBBNetworkManagerDownloadCompletionBlock downloadCompletion = [self completionBlockForDownload:dlTask];
+            [self setCompletionBlock:taskCompletion forTask:resumeTask];
+            [self setCompletionBlock:downloadCompletion forDownload:resumeTask];
+            [self removeCompletionBlockForTask:dlTask];
+            [self removeCompletionBlockForDownload:dlTask];
+            
+            [resumeTask resume];
+            return;
```

Comment:
This is the kind of early exit where I really like the Swift `guard`. Could you extract this into a method to make the early exit a little more obvious? I worry that it could get lost in a merge or other kind of "cleanup".